libp2p/multistream.nim

```diff
@@ -74,16 +74,21 @@ proc select*(m: MultistreamSelect,
     trace "successfully selected ", proto = proto[0]
     return
 
-  let protos = proto[1..<proto.len()]
-  trace "selecting one of several protos", protos = protos
-  for p in protos:
-    trace "selecting proto", proto = p
-    await conn.writeLp((p & "\n")) # select proto
-    result = string.fromBytes(await conn.readLp(1024)) # read the first proto
-    result.removeSuffix("\n")
-    if result == p:
-      trace "selected protocol", protocol = result
-      break
+  if proto.len > 1:
+    # Try to negotiate alternatives
+    let protos = proto[1..<proto.len()]
+    trace "selecting one of several protos", protos = protos
+    for p in protos:
+      trace "selecting proto", proto = p
+      await conn.writeLp((p & "\n")) # select proto
+      result = string.fromBytes(await conn.readLp(1024)) # read the first proto
```

Comment:
This can potentially return an invalid protocol - whatever was read on the last iteration, if we couldn't match a proto it should always return `""`.

Reply:
ah true, the else is screwing that up! nice catch 😥